### DISCUSSION TITLE
Does Micronaut maintain the Bean lifecycle for Prototype and default scopes

### DISCUSSION THREAD

**Discussion by qianmei** [micronaut-projects/micronaut-core#10414] [Q&A]
### Issue description

Hi,

Q1: How Micronaut handles the bean lifecycle for Prototype and default scopes? Any Docs? I read these docs [https://docs.micronaut.io/latest/guide/#scopes](url) and [https://docs.micronaut.io/latest/guide/#lifecycle](url)

Q2: If a bean gets created with Prototype and default scopes, will they get garbage collected by Micronaut? 
For example: a client who creates the beans by calling `ApplicationContext.getBean` or `ApplicationContext.createBean`, does the client have to explicitly call `ApplicationContext.destroyBean` to destroy the beans?

Thanks!

**Comment by sdelamo** [comment]
we have a scope guides: https://guides.micronaut.io/latest/micronaut-scope-types.html

**Comment by qianmei** [comment]
> we have a scope guides: https://guides.micronaut.io/latest/micronaut-scope-types.html

Thanks for sharing the link! 

I understand that there are multiple scopes types, I want to know that will Prototype and default scope beans get destroyed by Micronaut? Or I will need to explicitly call `ApplicationContext.destroyBean` to destroy the beans.

**Comment by graemerocher** [comment]
you have the explicitly destroy it yourself for prototypes. There is no way for Micronaut to know when a bean is due to be destroyed

**Comment by dstepanov** [comment]
It is destroyed when the parent is destroyed, when you get it directly from the bean context you need to destroy it yourself
